Hints.py

```diff
@@ -557,7 +564,21 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
 
     world.distribution.configure_gossip(spoiler, stoneIDs)
 
-    random.shuffle(stoneIDs)
+    stoneGroups = []
+    if 'groups' in world.hint_dist_user:
+        for group_names in world.hint_dist_user['groups']:
+            group = []
+            for stone_name in group_names:
```

Comment:
This is a complicated loop. We should just have a reverse map to look up from, and do that at top level so we don't have to recreate it for every world.

Reply:
That's a good point. I was originally just going to have the groups defined by ID, but figured that that would be pretty horrible to do by hand, so this was a kinda last second addition to reverse it. Part of the reason this loop is problematic is because any remaining ungrouped locations need to be put into their own single element lists and added in. Wasn't sure the cleanest way to figure that out.

Reply:
Cool, this look good.